node/src/components/block_accumulator/block_acceptor.rs

```diff
@@ -400,6 +410,26 @@ impl BlockAcceptor {
     }
 }
 
+/// Returns an error if the peer has sent too many finality signatures.
+fn check_signatures_from_peer_bound(
+    limit: u32,
+    peer: NodeId,
+    signatures: &BTreeMap<PublicKey, (FinalitySignature, BTreeSet<NodeId>)>,
+) -> Result<(), AcceptorError> {
+    let signatures_for_peer = signatures
+        .values()
+        .filter(|(_fin_sig, nodes)| nodes.contains(&peer))
+        .count();
+    let in_bound = u32::try_from(signatures_for_peer)
+        .map_or(false, |signatures_for_peer| signatures_for_peer < limit);
```

Comment:
Wouldn't it make more sense to do `signatures_for_peer < limit as usize`? No need for a `try_from` then.

Reply:
Or maybe I can do the opposite: `usize::try_from(limit)`?

Reply:
I did the conversion the other way around: `u32` → `usize`.